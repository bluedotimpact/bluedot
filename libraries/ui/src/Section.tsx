import React from 'react';
import clsx from 'clsx';

type BaseProps = {
  className?: string;
  title?: React.ReactNode;
  titleLevel?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  subtitle?: React.ReactNode;
  subtitleLevel?: 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
};

export type SectionHeadingProps = BaseProps;
export type SectionProps = React.PropsWithChildren<BaseProps>;

export const SectionHeading: React.FC<BaseProps> = ({
  title, titleLevel = 'h2', subtitle, subtitleLevel = 'p', className,
}) => {
  if (!title && !subtitle) return null;

  const HeadingTag = titleLevel;
  const SubtitleTag = subtitleLevel;
  return (
    <div className={clsx('section-heading__title-container flex justify-between items-center gap-space-between mb-6', className)}>
      <div className="section-heading__content flex-1 flex flex-col gap-2">
        {title && (
          <HeadingTag className="section-heading__title relative">
            {title}
          </HeadingTag>
        )}
        {subtitle && (
          <SubtitleTag className="section-heading__subtitle text-bluedot-darker text-md">{subtitle}</SubtitleTag>
        )}
      </div>
    </div>
  );
};

export const Section: React.FC<SectionProps> = ({
  className, title, titleLevel = 'h2', subtitle, subtitleLevel = 'p', children,
}) => {
  return (
    <section className={clsx('section section-body', className)}>
      <SectionHeading
        title={title}
        titleLevel={titleLevel}
        subtitle={subtitle}
        subtitleLevel={subtitleLevel}
      />
      <div className="section__body">
        {children}
      </div>
    </section>
  );
};
