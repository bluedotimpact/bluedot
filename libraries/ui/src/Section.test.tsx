import {
  describe, expect, test, afterEach,
} from 'vitest';
import { render, cleanup } from '@testing-library/react';
import { Section } from './Section';

describe('Section', () => {
  afterEach(() => {
    cleanup();
  });

  test('renders default as expected', () => {
    const { container } = render(<Section title="This is the title" />);
    expect(container).toMatchSnapshot();
  });
  test('renders with all optional props', () => {
    const { container, getByText } = render(
      <Section
        title="This is the title"
        subtitle="This is the subtitle"
      >
        <p>This is the yield</p>
      </Section>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('This is the title')).not.toBeNull();
    expect(getByText('This is the subtitle')).not.toBeNull();
    expect(getByText('This is the yield')).not.toBeNull();
  });
});
