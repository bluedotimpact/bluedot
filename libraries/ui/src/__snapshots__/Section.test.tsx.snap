// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Section > renders default as expected 1`] = `
<div>
  <section
    class="section section-body"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative"
        >
          This is the title
        </h2>
      </div>
    </div>
    <div
      class="section__body"
    />
  </section>
</div>
`;

exports[`Section > renders with all optional props 1`] = `
<div>
  <section
    class="section section-body"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative"
        >
          This is the title
        </h2>
        <p
          class="section-heading__subtitle text-bluedot-darker text-md"
        >
          This is the subtitle
        </p>
      </div>
      <div
        class="right-node"
      >
        This is the rightNode
      </div>
    </div>
    <div
      class="section__body"
    >
      <p>
        This is the yield
      </p>
    </div>
  </section>
</div>
`;
